lib/index.d.ts

```diff
@@ -0,0 +1,43 @@
+import {Request} from 'express';
+import * as passport from 'passport';
+
+export interface StrategyOptions {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback?: true;
+  scope?: string[];
+}
+
+export interface StrategyOptionsWithRequest {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback: true;
+  scope?: string[];
+}
+
+export interface VerifyOptions {
+  message: string;
+}
+
+export interface VerifyCallback {
+  (error: any, user?: any, options?: VerifyOptions): void;
+}
+
+export interface VerifyFunctionWithRequest {
+  (req: Request, accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): void;
+}
+
+export interface VerifyFunction {
+  (accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): void;
+}
+
+declare class Strategy implements Strategy {
+  public name: string;
+  public authenticate: (req: Request, options?: object) => void;
+
+  constructor(options: StrategyOptionsWithRequest, verify: VerifyFunctionWithRequest);
+  constructor(options: StrategyOptions, verify: VerifyFunction);
+  constructor(verify: VerifyFunction);
```

Comment:
This section adds a constructor with three different signatures. The first one takes the strategy options and passes the request to the verify function, the second one is strategy options and the verify function does _not_ get the express request object, and the final one is the verify function with no module options.

TypeScript will automatically select the appropriate one when building, this is much like "overrides" in C# or Java if you're familiar with those.